File: Go/334. Increasing Triplet Subsequence.go
package main

import "math"

//goland:noinspection GoUnusedFunction
func increasingTriplet(nums []int) bool {
	length := len(nums)

	if length < 3 {
		return false
	}

	first, second := math.MaxInt32, math.MaxInt32

	for _, num := range nums {
		if num <= first {
			first = num
		} else if num <= second {
			second = num
		} else {
			return true
		}
	}
	return false
}

File: TypeScript/334. Increasing Triplet Subsequence.ts
// noinspection JSUnusedGlobalSymbols
const increasingTriplet = (nums: number[]): boolean => {
  const length = nums.length

  if (length < 3) {
    return false
  }

  let first = Number.MAX_SAFE_INTEGER
  let second = first

  for (let i = 0; i < length; i++) {
    if (nums[i] <= first) {
      first = nums[i]
    } else if (nums[i] <= second) {
      second = nums[i]
    } else {
      return true
    }
  }
  return false
}
